src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4279,6 +4279,50 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
 
                     var expanded = memberResolutionResult.Result.Kind == MemberResolutionKind.ApplicableInExpandedForm;
                     var argsToParamsOpt = memberResolutionResult.Result.ArgsToParamsOpt;
+
+                    if (constructor is SynthesizedPrimaryConstructor primaryConstructor && primaryConstructor.GetCapturedParameters().Any())
+                    {
+                        for (int i = 0; i < analyzedArguments.Arguments.Count; i++)
+                        {
+                            if (analyzedArguments.RefKind(i) is (RefKind.Ref or RefKind.Out))
+                            {
+                                continue;
+                            }
+
+                            BoundParameter boundParameter;
+
+                            switch (analyzedArguments.Argument(i))
+                            {
+                                case BoundParameter param:
+                                    boundParameter = param;
+                                    break;
+
+                                case BoundConversion { Conversion.IsIdentity: true, Operand: BoundParameter param }:
+                                    boundParameter = param;
+                                    break;
+
+                                default:
+                                    continue;
+                            }
+
+                            if (boundParameter.ParameterSymbol is { } parameter &&
+                                primaryConstructor.GetCapturedParameters().ContainsKey(parameter))
+                            {
+                                if (expanded)
+                                {
+                                    var baseParameter = GetCorrespondingParameter(i, resultMember.Parameters, argsToParamsOpt, expanded: true);
+
+                                    if (baseParameter.Ordinal == resultMember.ParameterCount - 1 && OverloadResolution.IsValidParamsParameter(baseParameter))
```

Comment:
>if (baseParameter.Ordinal == resultMember.ParameterCount - 1 && OverloadResolution.IsValidParamsParameter(baseParameter)) [](http://example.com/codeflow?start=36&length=121)

Are we testing the case where this condition is not true? That is, an argument passed to an expanded call, but not passed to the `params` parameter.

Reply:
> Are we testing the case where this condition is not true? That is, an argument passed to an expanded call, but not passed to the `params` parameter.

Added a couple